### ISSUE TITLE
storageEngine opt of replSet is ignored

### ISSUE THREAD

**Issue by sebastian-nowak** [nodkz/mongodb-memory-server#86]
`storageEngine` option on `replSet` seems to be ignored:

```js
new MongoMemoryReplSet({
  replSet: {
    storageEngine: 'wiredTiger'
  }
})
```

It falls back to the default memory storage.

**Comment by AJRdev** [comment]
@nodkz I've tried to found the source of this issue, I might have found it but I don't understand this portion of the code in the function **getInstanceOpts** in the **MongoMemoryReplSet** class : 
```javascript
 getInstanceOpts(baseOpts: MongoMemoryInstancePropBaseT = {}): MongoMemoryInstancePropT {
    const rsOpts: ReplSetOpts = this.opts.replSet;
    const opts: MongoMemoryInstancePropT = {
      auth: !!rsOpts.auth,
      args: rsOpts.args,
      dbName: rsOpts.dbName,
      ip: rsOpts.ip,
      replSet: rsOpts.name,
    };
    if (baseOpts.args) opts.args = rsOpts.args.concat(baseOpts.args);
    if (baseOpts.port) opts.port = baseOpts.port;
    if (baseOpts.dbPath) opts.dbPath = baseOpts.dbPath;
    if (baseOpts.storageEngine) opts.storageEngine = baseOpts.storageEngine;
    this.debug('   instance opts:', opts);
    return opts;
  }
```
Why do we only extract the storageEngine from the baseOpts and not from the rsOpts as well like the args,dbName, etc ?
If I add this line in the construction of the opts : 
```javascript
    const opts: MongoMemoryInstancePropT = {
      auth: !!rsOpts.auth,
      args: rsOpts.args,
      dbName: rsOpts.dbName,
      ip: rsOpts.ip,
      replSet: rsOpts.name,
      storageEngine: rsOpts.storageEngine,
    };
```
The specified storageEngine specified in the constructor of **MongoMemoryReplSet** is then used and not the default one.

**Comment by AJRdev** [comment]
@jloveridge You might have an explanation as well on this behaviour since you have written the majority of the code of the MongoMemoryReplSet class :)

**Comment by AJRdev** [comment]
Issue fixed by the PRs #147 and #145
